src/gpu/nvidia/cudnn_matmul_lt.hpp

```diff
@@ -0,0 +1,521 @@
+/*******************************************************************************
+* Copyright 2020-2024 Intel Corporation
+* Copyright 2020 Codeplay Software Limited
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_NVIDIA_CUDNN_MATMUL_LT_HPP
+#define GPU_NVIDIA_CUDNN_MATMUL_LT_HPP
+
+#include "gpu/nvidia/cudnn_matmul_base.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace nvidia {
+
+struct cudnn_matmul_lt_t : cudnn_matmul_base_t {
+    using cudnn_matmul_base_t::cudnn_matmul_base_t;
+
+    struct pd_t : public gpu_matmul_pd_t {
+        using gpu_matmul_pd_t::gpu_matmul_pd_t;
+
+        DECLARE_COMMON_PD_T("cuda:cublaslt:any", cudnn_matmul_lt_t);
+
+        status_t init(impl::engine_t *engine) {
+            using namespace data_type;
+            using smask_t = primitive_attr_t::skip_mask_t;
+
+            data_type_t src_dt = src_md()->data_type;
+            data_type_t dst_dt = dst_md()->data_type;
+            data_type_t wei_dt = weights_md(0)->data_type;
+            data_type_t bia_dt
+                    = with_bias() ? weights_md(1)->data_type : data_type::f32;
+
+            bool f32_case = utils::everyone_is(f32, src_dt, wei_dt, dst_dt);
+            bool f16_case = utils::everyone_is(f16, src_dt, wei_dt, dst_dt);
+            bool bf16_case = utils::everyone_is(bf16, src_dt, wei_dt, dst_dt);
+#ifdef DNNL_NO_IMMA_INT8_DST
+            bool s8_case = utils::everyone_is(s8, src_dt, wei_dt)
+                    && utils::one_of(dst_dt, s32);
+#else
+            bool s8_case = false;
+            if (has_imma_dst_int8_support()) {
+                s8_case = utils::everyone_is(s8, src_dt, wei_dt)
+                        && utils::one_of(dst_dt, s8, s32);
+            } else {
+                s8_case = utils::everyone_is(s8, src_dt, wei_dt)
+                        && utils::one_of(dst_dt, s32);
+            }
+#endif
```

Comment:
Would it make sense to use a standard `if` instead of macro pre-processing?
This could help validation as there would be a single build to validate.